src/StaticWebAssetsSdk/Tasks/Data/StaticWebAsset.cs

```diff
@@ -549,7 +547,142 @@ public static string ComputeAssetRelativePath(ITaskItem asset, out string metada
             return asset.ItemSpec;
         }
 
-        public override bool Equals(object obj) => obj is StaticWebAsset asset && Equals(asset);
+        // Compares all fields in this order
+        // Identity
+        // SourceType
+        // SourceId
+        // ContentRoot
+        // BasePath
+        // RelativePath
+        // AssetKind
+        // AssetMode
+        // AssetRole
+        // AssetMergeSource
+        // AssetMergeBehavior
+        // RelatedAsset
+        // AssetTraitName
+        // AssetTraitValue
+        // Fingerprint
+        // Integrity
+        // CopyToOutputDirectory
+        // CopyToPublishDirectory
+        // OriginalItemSpec
+
+        public int CompareTo(StaticWebAsset other)
```

Comment:
Were all the new `IComparable` implementations added to make testing deterministic? Or is there another purpose for them?

Reply:
For testing and for ensuring that we use the same ordering when we write things into files.